exp/ses/delivery_notification.go

```diff
@@ -30,22 +32,54 @@ const (
 	COMPLAINT_FEEDBACK_TYPE_NOT_SPAM     = "not-spam"
 	COMPLAINT_FEEDBACK_TYPE_OTHER        = "other"
 	COMPLAINT_FEEDBACK_TYPE_VIRUS        = "virus"
+
+	// http://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-notifications-contents.html#receiving-email-notifications-contents-dkimverdict-object
+	VERDICT_STATUS_PASS              = "PASS"
+	VERDICT_STATUS_FAIL              = "FAIL"
+	VERDICT_STATUS_GRAY              = "GRAY"
+	VERDICT_STATUS_PROCESSING_FAILED = "PROCESSING_FAILED"
+
+	RECEIPT_ACTION_S3        = "S3"
+	RECEIPT_ACTION_SNS       = "SNS"
+	RECEIPT_ACTION_BOUNCE    = "Bounce"
+	RECEIPT_ACTION_LAMBDA    = "Lambda"
+	RECEIPT_ACTION_STOP      = "Stop"
+	RECEIPT_ACTION_WORK_MAIL = "WorkMail"
 )
 
 type SNSNotification struct {
 	NotificationType string     `json:"notificationType"`
-	Bounce           *Bounce    `json:"bounce" optional`
-	Complaint        *Complaint `json:"complaint" optional`
-	Delivery         *Delivery  `json:"delivery" optional`
+	Bounce           *Bounce    `json:"bounce,omitempty"`
+	Complaint        *Complaint `json:"complaint,omitempty"`
+	Delivery         *Delivery  `json:"delivery,omitempty"`
```

Comment:
I've replaced the `optional` with json's `omitempty` tag. The change was committed by mistake. If the `optional` tag is useful, please let me know, I'll revert it.